system-contract-dapp-playground/src/api/hedera/tokenTransfer-interactions/index.ts

```diff
@@ -0,0 +1,330 @@
+    console.error(err);
+    return { err, transactionHash: err.receipt && err.receipt.hash };
+  }
+};
+
+/**
+ * @dev transfers Hedera non-fungible tokens
+ *
+ * @dev integrates TokenTransferContract.transferNFTsPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param senders: string[]
+ *
+ * @param receivers: string[]
+ *
+ * @param serialNumbers: number[]
+ *
+ * @return Promise<TokenTransferSmartContractResult>
+ */
+export const transferNonFungibleTokens = async (
+  baseContract: Contract,
+  hederaTokenAddress: string,
+  senders: string[],
+  receivers: string[],
+  serialNumbers: number[]
+): Promise<TokenTransferSmartContractResult> => {
+  // sanitize params
+  let sanitizeErr;
+  if (!isAddress(hederaTokenAddress)) {
+    sanitizeErr = 'Invalid token address';
+  }
+  if (!sanitizeErr) {
+    senders.some((address) => {
+      if (!isAddress(address)) {
+        sanitizeErr = `${address} is an invalid sender accountID`;
+        return true;
+      }
+    });
+  }
+  if (!sanitizeErr) {
+    receivers.some((address) => {
+      if (!isAddress(address)) {
+        sanitizeErr = `${address} is an invalid receiver accountID`;
+        return true;
+      }
+    });
+  }
+  if (!sanitizeErr) {
+    serialNumbers.some((seriNum) => {
+      if (seriNum < 0) {
+        sanitizeErr = `${seriNum} is an invalid serial number`;
+        return true;
+      }
+    });
+  }
+  if (sanitizeErr) {
+    console.error(sanitizeErr);
+    return { err: sanitizeErr };
+  }
+
+  // invoking contract methods
+  try {
+    const tx = await baseContract.transferNFTsPublic(
+      hederaTokenAddress,
+      senders,
+      receivers,
+      serialNumbers
+    );
+
+    return await handleContractResponse(tx);
+  } catch (err: any) {
+    console.error(err);
+    return { err, transactionHash: err.receipt && err.receipt.hash };
+  }
+};
+
+/**
+ * @dev transfers single token (fungible vs non-fungible)
+ *
+ * @dev integrates TokenTransferContract.transferTokenPublic()
+ *
+ * @dev integrates TokenTransferContract.transferNFTPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param API: "FUNGIBLE" | "NON_FUNGIBLE"
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param sender: string
+ *
+ * @param receiver: string
+ *
+ * @param quantity: number (amount/serialNumber)
+ *
+ * @return Promise<TokenTransferSmartContractResult>
+ */
+export const transferSingleToken = async (
+  baseContract: Contract,
+  API: 'FUNGIBLE' | 'NON_FUNGIBLE',
+  hederaTokenAddress: string,
+  sender: string,
+  receiver: string,
+  quantity: number
+): Promise<TokenTransferSmartContractResult> => {
+  // sanitize params
+  let sanitizeErr;
+  if (!isAddress(hederaTokenAddress)) {
+    sanitizeErr = 'Invalid token address';
+  } else if (!isAddress(sender)) {
+    sanitizeErr = 'Invalid sender address';
+  } else if (!isAddress(receiver)) {
+    sanitizeErr = 'Invalid receiver address';
+  } else if (quantity < 0) {
```

Comment:
Q: should we add a max for safe guarding? Check the protobuf and see if there's a max we can utilize.